### ISSUE TITLE
[QST] Optimal Size of Parquet Part Files

### ISSUE THREAD

**Issue by silpara** [NVIDIA-Merlin/NVTabular#1607]
**Between small sized, large number of part parquet files vs large sized, fewer part parquet files, which one is better for NVTabular's data loading performance on GPU?**

When writing parquet files from spark jobs, there is a choice to either repartition or coalesce to desired number of partitions each eventually written as part files. This also dictates the size of each of the files. For NVTabular's performance how to find the optimal size and number of parquet files?

**Comment by rjzamora** [comment]
The current `Dataset` API (used by the dataloaders) will not aggregate data from multiple files into the same Dask-DataFrame partition. Therefore, the dataloaders will not fetch data from multiple files at once (yet).  **An optimial dataset layout will typically comprise a small number of Parquet files with many row-groups in each file.** At a minimum, a single Parquet file should contain enough records to populate 1/8 of a single-GPU's total memory (as a `cudf.DataFrame` object). Depending on the GPU and Parquet compression, this can be anywhere between 10MB and 10GB on disk.

With that said,  it is quite common to have Parquet files <1GB in practice. Therefore, I do believe the Dataset API should be improved in the near future to support more-efficient multi-file partitioning.

In summary, you are probably best off with a smaller number of large Parquet files for now.  However, this recomendation could change in the future if/when file-aggregation is improved. (we would certainly try to announce such a change loudly)
